mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/index/MongoDBIndexOptionsConflictsTest.java

```diff
@@ -0,0 +1,279 @@
+
+		UniqueConstraintAndUniqueIndexEntity entity = new UniqueConstraintAndUniqueIndexEntity();
+		entity.name = "Fabio Massimo";
+		entity.email = "fabiomassimo@myemailprovider.eu";
+		entity.address = "11bis Rue Roquepine Paris";
+
+		try ( SessionFactory sessionFactory = config.build() ) {
+			inTransaction( sessionFactory, session -> {
+				session.persist( entity );
+			} );
+
+			inTransaction( sessionFactory, session -> {
+				UniqueConstraintAndUniqueIndexEntity stored = session.load( UniqueConstraintAndUniqueIndexEntity.class, entity.name );
+				assertThat( entity.email ).isEqualTo( stored.getEmail() );
+				assertThat( entity.address ).isEqualTo( stored.getAddress() );
+			} );
+
+			inTransaction( sessionFactory, session -> {
+				session.delete( entity );
+			} );
+		}
+	}
+
+	@Test
+	public void testUniqueIndexAndNOTUniqueIndex() {
+		SessionFactoryBuilder config = SessionFactoryBuilder.entities( UniqueIndexAndNotUniqueIndexEntity.class );
+
+		UniqueIndexAndNotUniqueIndexEntity entity = new UniqueIndexAndNotUniqueIndexEntity();
+		entity.name = "Fabio Massimo";
+		entity.email = "fabiomassimo@myemailprovider.eu";
+		entity.address = "11bis Rue Roquepine Paris";
+
+		try ( SessionFactory sessionFactory = config.build() ) {
+			inTransaction( sessionFactory, session -> {
+				session.persist( entity );
+			} );
+
+			inTransaction( sessionFactory, session -> {
+				UniqueIndexAndNotUniqueIndexEntity stored = session.load( UniqueIndexAndNotUniqueIndexEntity.class, entity.name );
+				assertThat( entity.email ).isEqualTo( stored.getEmail() );
+				assertThat( entity.address ).isEqualTo( stored.getAddress() );
+			} );
+
+			inTransaction( sessionFactory, session -> {
+				session.delete( entity );
+			} );
+		}
+	}
+
+	@Test
+	public void testTwoUniqueIndexes() {
+		SessionFactoryBuilder config = SessionFactoryBuilder.entities( TwoUniqueIndexesEntity.class );
+
+		TwoUniqueIndexesEntity entity = new TwoUniqueIndexesEntity();
+		entity.name = "Fabio Massimo";
+		entity.email = "fabiomassimo@myemailprovider.eu";
+		entity.address = "11bis Rue Roquepine Paris";
+
+		try ( SessionFactory sessionFactory = config.build() ) {
+			inTransaction( sessionFactory, session -> {
+				session.persist( entity );
+			} );
+
+			inTransaction( sessionFactory, session -> {
+				TwoUniqueIndexesEntity stored = session.load( TwoUniqueIndexesEntity.class, entity.name );
+				assertThat( entity.email ).isEqualTo( stored.getEmail() );
+				assertThat( entity.address ).isEqualTo( stored.getAddress() );
+			} );
+
+			inTransaction( sessionFactory, session -> {
+				session.delete( entity );
+			} );
+		}
+	}
+
+	@Test
+	public void testNormalIndexAndFullTextSearchIndex() {
+		SessionFactoryBuilder config = SessionFactoryBuilder.entities( NormalIndexAndFullTextSearchIndexEntity.class );
+
+		try ( SessionFactory ignored = config.build() ) {
+			fail( "Expected exception at Hibernate factory creation time was not raised!" );
```

Comment:
Let's change the message to:
```
Should fail  because we defined two index with conflicting configurations on field `UniqueConstraintAndNotUniqueIndexEntity#email`
```
Or whatever the reason you expect a failure

Reply:
s/index/indexes/